proposed/2023/release-staging-and-deprecation.md

```diff
@@ -0,0 +1,199 @@
+# ***Release staging and deprecation***
+<!-- Replace `Title` with an appropriate title for your design -->
+
+- Author Name Artak Mkrtchyan ([mkArtakMSFT](https://github.com/mkArtakMSFT))
+- GitHub Issue <https://github.com/NuGet/NuGetGallery/issues/3931>
+
+## Summary
+
+This proposal introduces a concept for grouping packages that are published to NuGet.
```

Comment:
First of all thanks for bringing up this topic and working on the spec. This looks pretty thought through including multiple scenarios. And big +1 for making this out of the package metadata, which looked like a misfit for this usecase. Most of my subsequent comments might look towards simplifying the experience for MVP. So do bear with me 😄